src/main/java/com/uber/jenkins/phabricator/conduit/ConduitAPIClient.java

```diff
@@ -0,0 +1,109 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.uber.jenkins.phabricator.conduit;
+
+import net.sf.json.JSONObject;
+import net.sf.json.groovy.JsonSlurper;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ConduitAPIClient {
+    private static final String API_TOKEN_KEY = "api.token";
+
+    private final String conduitURL;
+    private final String conduitToken;
+
+    public ConduitAPIClient(String conduitURL, String conduitToken) {
+        this.conduitURL = conduitURL;
+        this.conduitToken = conduitToken;
+    }
+
+    /**
+     * Call the conduit API of Phabricator
+     * @param action Name of the API call
+     * @param data The data to send to Harbormaster
+     * @return The result as a JSONObject
+     * @throws IOException If there was a problem reading the response
+     * @throws ConduitAPIException If there was an error calling conduit
+     */
+    public JSONObject perform(String action, Map<String, String> data) throws IOException, ConduitAPIException {
+        CloseableHttpClient client = HttpClientBuilder.create().build();
+        HttpUriRequest request = createRequest(action, data);
+
+        HttpResponse response;
+        try {
+            response = client.execute(request);
+        } catch (ClientProtocolException e) {
+            throw new ConduitAPIException(e.getMessage());
+        }
+
+        InputStream responseBody = response.getEntity().getContent();
+
+        if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+            throw new ConduitAPIException(responseBody.toString(), response.getStatusLine().getStatusCode());
+        }
+
+        JsonSlurper jsonParser = new JsonSlurper();
+        return (JSONObject)jsonParser.parse(responseBody);
+    }
+
+    public HttpUriRequest createRequest(String action, Map<String, String> data) throws UnsupportedEncodingException, ConduitAPIException {
+        HttpPost post;
+        try {
+            post = new HttpPost(
```

Comment:
I had that originally, but when testing manually I was running into issues with double-slashes, e.g. "http://phabricator.testing//api/" when the user inputted a trailing slash. Rather than doing manual string munging, I thought I'd just use URL, but I agree that this is awkward. Is there a better way?